pypeit/scripts/show_2dspec.py

```diff
@@ -127,77 +131,153 @@ def main(args):
 
-        # Grab the slit edges
-        slits = spec2DObj.slits
-        if spec2DObj.sci_spat_flexure is not None:
-            msgs.info("Offseting slits by {}".format(spec2DObj.sci_spat_flexure))
-        all_left, all_right, mask = slits.select_edges(flexure=spec2DObj.sci_spat_flexure)
-
-        # TODO -- This may be too restrictive, i.e. ignore BADFLTCALIB??
-        gpm = mask == 0
+        # Try to read the Spec2DObj using the current datamodel, but allowing
+        # for the datamodel version to be different
+        try:
+            spec2DObj = spec2dobj.Spec2DObj.from_file(args.file, detname, chk_version=False)
+        except PypeItError:
+            try:
+                # Try to get the pypeit version used to write this file
+                file_pypeit_version = fits.getval(args.file, 'VERSPYP', 0)
+            except KeyError:
+                file_pypeit_version = '*unknown*'
+            msgs.warn(f'Your installed version of PypeIt ({__version__}) cannot be used to parse '
+                      f'{args.file}, which was reduced using version {file_pypeit_version}.  You '
+                      'are strongly encouraged to re-reduce your data using this (or, better yet, '
+                      'the most recent) version of PypeIt.  Script will try to parse only the '
+                      'relevant bits from the spec2d file and continue (possibly with more '
+                      'limited functionality).')
+            spec2DObj = None
+
+        if spec2DObj is None:
+            # Try to get the relevant elements directly from the fits file
+            with io.fits_open(args.file) as hdu:
+                names = [h.name for h in hdu]
+                has_det = any([detname in n for n in names])
+                if not has_det:
+                    msgs.error(f'Provided file has no extensions including {detname}.')
+                for ext in ['SCIIMG', 'SKYMODEL', 'OBJMODEL', 'IVARMODEL']:
+                    _ext = f'{detname}-{ext}'
+                    if _ext not in names:
+                        msgs.error(f'{args.file} missing extension {_ext}.')
+
+                sciimg = hdu[f'{detname}-SCIIMG'].data
+                skymodel = hdu[f'{detname}-SKYMODEL'].data
+                objmodel = hdu[f'{detname}-OBJMODEL'].data
+                ivarmodel = hdu[f'{detname}-IVARMODEL'].data
+                _ext = f'{detname}-BPMMASK'
+                bpmmask = hdu[_ext].data if _ext in names else None
+                _ext = f'{detname}-WAVEIMG'
+                waveimg = hdu[_ext].data if _ext in names else None
+
+                img_gpm = np.ones(sciimg.shape, dtype=bool) if bpmmask is None else bpmmask == 0
+                model_gpm = img_gpm.copy()
+
+                _ext = f'{detname}-SLITS'
+                if _ext not in names:
+                    msgs.warn(f'{args.file} missing extension {_ext}; cannot show slit edges.')
+                else:
+                    slit_columns = hdu[_ext].columns.names
```

Comment:
same comment as above.

and, boy, this is painful to look at ;)

Reply:
Same response.  And, yes, agreed, although not as painful as `DataContainer` :)

Reply:
I agree, but it is not as painful as not being able to view your reduced data.